src/OrchardCore.Modules/OrchardCore.Https/Startup.cs

```diff
@@ -43,17 +35,28 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddScoped<INavigationProvider, AdminMenu>();
             services.AddScoped<IDisplayDriver<ISite>, HttpsSettingsDisplayDriver>();
             services.AddSingleton<IHttpsService, HttpsService>();
-            services.AddSingleton(new RewriteOptions());
-            
-            services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<RewriteOptions>, RewriteOptionsHttpsConfiguration>());
+
+            services.AddOptions<HttpsRedirectionOptions>()
+                .Configure<IHttpsService>((options, service) =>
+                {
+                    var settings = service.GetSettingsAsync().GetAwaiter().GetResult();
+                    if (settings.RequireHttpsPermanent)
+                    {
+                        options.RedirectStatusCode = StatusCodes.Status308PermanentRedirect;
+                    }
+
+                    if (settings.SslPort != null)
+                    {
+                        options.HttpsPort = settings.SslPort;
+                    }
+                });
 
             services.AddHsts(options =>
             {
                 options.Preload = false;
                 options.IncludeSubDomains = true;
```

Comment:
Could enabling subdomains by default catch anyone out as the default setting?

Reply:
Good remark! It's going to be a bit tricky to configure, as OC allows defining multiple domains and supports wildcards for subdomains:
  - If we blindly set it to `true`, it will potentially impact other tenants (e.g `tenant2.orchardsite.com`).
  - If we blindly set it to `false`, it won't protect subdomains, even if they technically belong to the same tenant.

Given that OC supports per-path tenants, we should also prevent this module from being enabled in non-default tenants so that HSTS settings don't leak across tenants.

/cc @sebastienros